src/snowflake/cli/api/cli_global_context.py

```diff
@@ -274,13 +277,36 @@ def project_root(self):
     def set_project_root(self, project_root: Path):
         self._project_root = project_root
 
+    @property
+    def project_path_arg(self) -> Optional[str]:
+        return self._project_path_arg
+
+    def set_project_path_arg(self, project_path_arg: str):
+        self._project_path_arg = project_path_arg
+
+    @property
+    def project_env_overrides_args(self) -> list[str]:
```

Comment:
Wouldn't it be better to use some kind of dictionary instead of list?

Reply:
these come directly from the cli. I can update the typer callback to transform them before storing them.